kolibri/content/utils/paths.py

```diff
@@ -1,13 +1,22 @@
 import os
+import re
 
 from django.conf import settings
+from django.core.urlresolvers import reverse
 
 try:
     from urlparse import urljoin
 except ImportError:
     from urllib.parse import urljoin
 
 
+# valid storage filenames consist of 32-char hex plus a file extension
+VALID_STORAGE_FILENAME = re.compile("[0-9a-f]{32}\.[0-9a-z]+")
```

Comment:
Ah, OK. Fair enough. Do you think listing all the extensions would be
awful?

On Thu, 8 Sep 2016, 17:19 Jamie Alexandre, <notifications@github.com> wrote:

> In kolibri/content/utils/paths.py
> <https://github.com/learningequality/kolibri/pull/515#discussion_r78112138>
> :
>
> >
> >  try:
> >      from urlparse import urljoin
> >  except ImportError:
> >      from urllib.parse import urljoin
> >
> >
> > +# valid storage filenames consist of 32-char hex plus a file extension
> > +VALID_STORAGE_FILENAME = re.compile("[0-9a-f]{32}\.[0-9a-z]+")
>
> We would need to make a full list of extensions, as this function using
> this variable handles locating all types of files, not just the zipped
> ones. The main reason for VALID_STORAGE_FILENAME is safety validation, to
> ensure there's no funny business with .. escaping out of the content
> directory, etc.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/learningequality/kolibri/pull/515/files/b49479a3cc26d690958dd5297f81ba7b4ebbbe4c#r78112138>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABmkvdY0kZSwkpuxmGpdHl_FpNcMT07sks5qoKYngaJpZM4J4dcC>
> .
>